data/domain/construction.tsx

```diff
@@ -165,6 +173,15 @@ export const updateConstruction = (data: Map<string, any>) => {
         if (!building.maxed) {
             building.finishedUpgrade = building.currentXP >= building.getBuildCost();
         }
+
+        building.buildPercentage = Math.min(Math.round((building.currentXP * 100) / building.getBuildCost()), 100);
+
+        building.upgradable = building.getLevelCosts(building.level, costCruncher).filter((costData, index) => {
+            const amountInStorage = storage?.amountInStorage(costData.item) ?? 0;
+            if (amountInStorage >= costData.quantity) {
+                return true;
+            }
+        }).length == 2 && (building.buildPercentage === 100 || building.nextLevelUnlocked);
```

Comment:
It's funny that the length == 2 works even though some buildings only have a single item for cost, but the game data puts the 2nd item as `Blank` with cost zero so since you are using `>=` I think it will be fine haha